### PR TITLE
[cmake] introduce builtin_openui5 option, default ON

### PR DESCRIPTION
**Pull request by linev**
1. Move openui5 package to `builtins/openui5` subdirectory
2. Let install full openui5 library version from web when builtin_openui5=OFF
    Can be required when users want to used other openui5 components which are not included in provided package.
3. For developers let configure `$ENV{OPENUI5DIR}` when just via symbolic link existing version can be used

### CODE REVIEW

**Comment by linev** [comment]
Looking into jenkins logs - it tries to build with `builtin_openui5=OFF`.
It works, but requires loading of full openui5 package from web.
After merging this PR one has to change Jenkins to enable `builtin_openui5`

**Review by oshadura — APPROVED** [review]
Definitely LGTM! Could please add these changes to the release notes so we will not forget about them? (as well about new option `builtin_nlohmannjson`)

File: cmake/modules/SearchInstalledSoftware.cmake [hunk 1]
```diff
@@ -1792,15 +1792,33 @@ endif()
...
-     URL_HASH SHA256=0cde25387047163fe2ed5f32eb7224628be10c3b6afa07d0b2a42002543909c2
-     CONFIGURE_COMMAND ""
-     BUILD_COMMAND ""
-     INSTALL_COMMAND ""
-     SOURCE_DIR ${CMAKE_BINARY_DIR}/ui5/distribution)
-  install(DIRECTORY ${CMAKE_BINARY_DIR}/ui5/distribution/ DESTINATION ${CMAKE_INSTALL_OPENUI5DIR}/distribution/ COMPONENT libraries FILES_MATCHING PATTERN "*")
+  if(NOT "$ENV{OPENUI5DIR}" STREQUAL "" AND EXISTS "$ENV{OPENUI5DIR}/resources/sap-ui-core-nojQuery.js")
+     message(STATUS "openui5 - use from $ENV{OPENUI5DIR}")
+     file(MAKE_DIRECTORY ${CMAKE_BINARY_DIR}/ui5)
+     execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink
```
Comment (oshadura) [inline cmake/modules/SearchInstalledSoftware.cmake:1798]: @linev just a small comment, can you please check here what will happen in case of make install?

Comment (linev) [reply cmake/modules/SearchInstalledSoftware.cmake:1798]: It is not dedicated for that - just keep possibility to tweak ROOT build procedure to use openui from other directory. Only for expert usage

Comment (oshadura) [reply cmake/modules/SearchInstalledSoftware.cmake:1798]: Let's add this explicitly in the status message and as well in Release notes :+1:

Comment (linev) [reply cmake/modules/SearchInstalledSoftware.cmake:1801]: I add few comments into SearchInstalledSoftware.cmake file

**Pull request merged**